addons/payment_adyen/controllers/main.py

```diff
@@ -29,3 +29,18 @@ def adyen_return(self, **post):
             custom = json.loads(post.pop('merchantReturnData', '{}'))
             return_url = custom.pop('return_url', '/')
         return werkzeug.utils.redirect(return_url)
+
+    @http.route([
+        '/payment/adyen/notification',
+    ], type='http', auth='none', methods=['POST'])
+    def adyen_notification(self, **post):
+        if post.get('eventCode') in ['AUTHORISATION']:
+            tx_id = request.registry['payment.transaction'].search(request.cr, SUPERUSER_ID, [('reference', 'in', [post.get('merchantReference')])], limit=1)
```

Comment:
Some remarks 
- forgot context ? (request.context)
- some flattening
    tx_id = None
    if post.get('eventCode') in [] and post.get('merchantReference'):
      tx_id = ...
   if not tx_id: return ...